fili-core/src/test/groovy/com/yahoo/bard/webservice/table/PermissiveConcretePhysicalTableSpec.groovy

```diff
@@ -0,0 +1,100 @@
+package com.yahoo.bard.webservice.table
+
+import static com.yahoo.bard.webservice.data.time.DefaultTimeGrain.DAY
+import static org.joda.time.DateTimeZone.UTC
+
+import com.yahoo.bard.webservice.data.config.names.TableName
+import com.yahoo.bard.webservice.data.dimension.BardDimensionField
+import com.yahoo.bard.webservice.data.dimension.Dimension
+import com.yahoo.bard.webservice.data.dimension.DimensionColumn
+import com.yahoo.bard.webservice.data.dimension.MapStoreManager
+import com.yahoo.bard.webservice.data.dimension.impl.KeyValueStoreDimension
+import com.yahoo.bard.webservice.data.dimension.impl.ScanSearchProviderManager
+import com.yahoo.bard.webservice.data.metric.MetricColumn
+import com.yahoo.bard.webservice.metadata.TestDataSourceMetadataService
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import org.joda.time.Interval
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class PermissiveConcretePhysicalTableSpec extends Specification {
+    @Shared PermissiveConcretePhysicalTable permissiveConcretePhysicalTable
+
+    @Shared TableName tableName
+
+    @Shared Set<Column> columns
+
+    @Shared Dimension dimension
+    @Shared DimensionColumn dimensionColumn
+    @Shared MetricColumn metricColumn1, metricColumn2
+
+    @Shared Interval interval1, interval2, interval3
+
+    def setupSpec() {
+        tableName = TableName.of('test table')
+
+        dimensionColumn = new DimensionColumn(
+                new KeyValueStoreDimension(
+                        "dimension",
+                        null,
+                        [BardDimensionField.ID] as LinkedHashSet,
+                        MapStoreManager.getInstance("dimension"),
+                        ScanSearchProviderManager.getInstance("apiProduct")
+                )
+        )
+        metricColumn1 = new MetricColumn("metric1")
+        metricColumn2 = new MetricColumn("metric2")
+
+        columns = [dimensionColumn, metricColumn1, metricColumn2] as Set
+
+        interval1 = new Interval("2014-07-03/2014-07-09")
+        interval2 = new Interval("2014-07-03/2014-07-05")
+        interval3 = new Interval("2014-07-05/2014-07-09")
+
+        permissiveConcretePhysicalTable = new PermissiveConcretePhysicalTable(
+                tableName,
+                DAY.buildZonedTimeGrain(UTC),
+                columns,
+                ['dimension': 'druidDim'],
+                new TestDataSourceMetadataService([
+                        (dimensionColumn): [interval1] as Set,
+                        (metricColumn1): [interval2] as Set,
+                        (metricColumn2): [interval3] as Set,
+                        (new Column('ignored')): [new Interval('2010-01-01/2500-12-31')] as Set
+                ])
+        )
+    }
+
+    @Unroll
+    def "Physical table getAvailableIntervals returns #expected for column #allColumnNames"() {
```

Comment:
It would be better to have the test name indicate _why_ or under what conditions the behavior being tested is true.

For example "The spanning intervals are returned even when individual columns are not fully covering the interval" might be a good name because it's clear what the behavior is, and under what conditions.